packages/amplify-graphql-relational-transformer/src/has-many/has-many-directive-transformer-factory.ts

```diff
@@ -23,14 +23,18 @@ export const getHasManyDirectiveTransformer = (
     case 'DYNAMODB':
       // If references are passed to the directive, we'll use the references relational
       // modeling approach.
-      if (config.references?.length >= 1) {
+      if (config.references) {
         // Passing both references and fields is not supported.
-        if (config.fields?.length > 0) {
+        if (config.fields) {
```

Comment:
same here: avoid nesting for known error conditions.

Reply:
https://github.com/aws-amplify/amplify-category-api/pull/2381/files#r1543221887